### DISCUSSION TITLE
Filter optional relation

### DISCUSSION THREAD

**Discussion by galkatz373** [prisma/prisma#5901] [Q&A] [answered]
Hey,

I have OrderRow model, which has an optional Makat relation.
When I do the query below the prisma returns only OrderRows which have Makat relation, even though I would like to get those without it as well.
If I'm looking at sql generated and I see inner joins instead of left joins.

Is there a workaround?

```js
return await prisma.orderRow.findMany({
          cursor: cursor
            ? {
                id: cursor,
              }
            : undefined,
          skip: cursor ? 1 : 0,
          orderBy: [
            {
              updateDate: "desc",
            },
            { id: "desc" },
          ],
          take: pageSize,
          where: {
            orderId,
            OR: [
              {
                makat: { code: { contains: filter } },
              },
              { makat: { name: { contains: filter } } },
              { code: { equals: +filter } },
              { amount: { equals: +filter } },
            ],
          },
        });
```

Thanks in advance!

**Comment by ryands17** [comment, marked as answer]
@galkatz373 👋 

As per the query, it seems like a 1-1 optional relation. In this case, you can do the following to get the `OrderRows` which do not have a `Makat`:

```ts
return await prisma.orderRow.findMany({
          cursor: cursor
            ? {
                id: cursor,
              }
            : undefined,
          skip: cursor ? 1 : 0,
          orderBy: [
            {
              updateDate: "desc",
            },
            { id: "desc" },
          ],
          take: pageSize,
          where: {
            orderId,
            OR: [
              {
                makat: { code: { contains: filter } },
              },
              { makat: { name: { contains: filter } } },
              { code: { equals: +filter } },
              { amount: { equals: +filter } },
              { makat: { is: null } },
            ],
          },
});
```

Let me know if this works. If not, you could share your `schema.prisma` so that I can give you the correct query.

**Reply by galkatz373** [reply]
Worked thank you very much!
It would be nice to have on the docs as well.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
